### PR TITLE
Install and compile with ccache for faster builds

### PR DESCRIPTION
**Pull request by prigoyal**
installing this will significantly speed up the build time

### CODE REVIEW

File: docker/common/install_ccache.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -ex
+
+# Install ccache from source.
+# Needs 3.4 or later for ccbin support
+pushd /tmp
+git clone https://github.com/ccache/ccache -b v3.4.1
```
Comment (ftynse) [inline docker/common/install_ccache.sh]: why can't we have it from binary repositories?  ubuntu has it.

Comment (ftynse) [reply docker/common/install_ccache.sh]: do we want some specific version?  why?

Comment (prigoyal) [reply docker/common/install_ccache.sh]: may I request that I go over the setup once using the recommended practice and then  I will be able to answer questions better. This PR is still marked as WIP. Thanks for cooperation. :)

Comment (prigoyal) [reply docker/common/install_ccache.sh]: to answer the question: trusty has ccache 3.1 version which doesn't work with nvcc well. The nvcc support was added in version 3.4.3 so we install this way.

**Comment by prigoyal** [comment]
Hi @abadams , @zdevito can you help review this PR? I am not aware of who else is using ccache on the team. I have tested the docker images with these and TC builds fine and tests fine. Thanks :)

**Comment by prigoyal** [comment]
thanks @zdevito :)

**Pull request merged**